src/combinatorics/simple_graph/basic.lean

```diff
@@ -204,6 +302,19 @@ lemma complete_graph_is_regular [decidable_eq V] :
   (complete_graph V).is_regular_of_degree (fintype.card V - 1) :=
 by { intro v, simp }
 
+/--
+The minimum degree of all vertices
+-/
+def min_degree (G : simple_graph V) [nonempty V] [decidable_rel G.adj] : ℕ :=
+finset.min' (univ.image (λ (v : V), G.degree v)) (nonempty.image univ_nonempty _)
```

Comment:
Does this also work as
```suggestion
finset.min' (univ.image G.degree) (nonempty.image univ_nonempty _)
```

Reply:
I suggested this too in the discord yesterday and it turned out not to work :)

Reply:
`G.degree` works for any vertex with a finite set of neighbors, and the proof of finiteness is an instance argument given after the vertex itself, so that's why it needs to be wrapped in a lambda expression.

(I'm wondering if it's useful talking about degrees in graphs that aren't locally finite.  Maybe
```lean
def degree' [∀ w, fintype (G.neighbor_set w)] (v : V) : ℕ := (G.neighbor_finset v).card
```
would be better.  I'm not suggesting that we change it right now, but it would be worth looking at this again if the `min_degree` use of `G.degree` is common.)